### PR TITLE
Refactor table alias semantics & fix parenthesized join construct

### LINKED ISSUE #612: [Postgres[ Parser does not understand table aliases inside join construct
Related to: https://github.com/tobymao/sqlglot/issues/423

The same error occurs as in the issue above when using table aliases inside the parentheses.

So this works (fixed in previous issue):
```
select * from (table1 left join table2 on 1 = 1);
```

But this doesn't:
```
select * from (table1 t1 left join table2 t2 on 1 = 1);
```

```
ParseError: Expecting ). Line 1, Col: 26.
  select * from (table1 t1 left join table2 t2 on 1 = 1)

Invalid expression / Unexpected token. Line 1, Col: 54.
  select * from (table1 t1 left join table2 t2 on 1 = 1)
```

Using sqlglot 8.0.2.

### PR DESCRIPTION
**Pull request by GeorgeSittas**
WIP (experimenting with changing table alias semantics...)
Fixes #612

### CODE REVIEW

**Comment by tobymao** [comment]
fyi @barakalon this is going to change how we represent alias's on tables. before we did 

alias ( table), but now alias will be a property in table

**Comment by barakalon** [comment]
I'm surprised the optimizer didn't explode with this. Nice.

@ginter I think this will simplify some of our code.

**Comment by tobymao** [comment]
@barakalon it does explode, @GeorgeSittas is fixing it now

**Comment by GeorgeSittas** [comment]
It does seem to simplify parts of the code indeed

File: sqlglot/parser.py [hunk 1]
```diff
@@ -1120,6 +1120,9 @@ def _parse_subquery(self, this):
         )
 
     def _parse_query_modifiers(self, this):
+        if isinstance(this, exp.Alias):
```
Comment (tobymao) [inline sqlglot/parser.py:1123]: i don't think this is needed?

File: tests/dialects/test_dialect.py [hunk 2]
```diff
@@ -716,6 +716,13 @@ def test_joined_tables(self):
                 "mysql": "SELECT * FROM (tbl1 JOIN LATERAL (SELECT * FROM bla) AS tbl)",
             },
         )
+        self.validate_all(
+            "SELECT * FROM (table1 t1 LEFT JOIN table2 t2 ON 1 = 1)",
```
Comment (tobymao) [inline tests/dialects/test_dialect.py:720]: i think you only need an identity sql test for this one

**Review by barakalon — APPROVED** [review]
Awesome!

This will be another breaking change 🙃

**Pull request merged**